docs/backup-recovery/README.md

```diff
@@ -0,0 +1,132 @@
+
+The following extra-manifest MachineConfig will create a 50G partition on the specified disk:
+
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-recovery-partition
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      disks:
+        # Use persistent disk path for device, as /dev/sd* names can change
+        - device: /dev/disk/by-path/pci-0000:18:00.0-scsi-0:2:1:0
+          wipeTable: true # Wipe the disk
+          partitions:
+          - label: recovery
+            startMiB: 1 # Optional
+            sizeMiB: 51200
+      filesystems:
+        - device: /dev/disk/by-partlabel/recovery
+          path: /var/recovery
+          format: xfs
+          wipe_filesystem: true
+    systemd:
+      units:
+        - name: var-recovery.mount
+          enabled: true
+          contents: |
+            [Unit]
+            Before=local-fs.target
+            [Mount]
+            What=/dev/disk/by-partlabel/recovery
+            Where=/var/recovery
+            Options=defaults
+            [Install]
+            WantedBy=local-fs.target
+```
+
+## Backup workload and the procedure ##
+
+Backup workload is a one-shot task created on each of the spoke cluster nodes to trigger backup and keep the backup and system resources in the recovery partition in order to recovery a failed upgrade. For SNO spokes it is realized as a `batch/v1 job`.
+
+
+### On the hub ###
+
+- User creates a TALO CR that defines:
+  - A set of clusters to be upgraded  
+  - The need for resource backup
+- User applies the TALO CR to the hub cluster at the beginning of the maintenance window
+- TALO checks the backup requirement in the TALO CR. If required, deploys a backup workload on the designated spoke.  
+
+### States ###
+
+- BackupStatePreparingToStart - is the initial state all clusters are automatically assigned to on the first reconciliation pass of the TALO CR. Upon entry TALO deletes spoke backup namespace and hub view resources that might have remained from the prior incomplete attempts.
+- BackupStateStarting - is the state for creation of the backup job pre-requisites and the job itself
+- BackupStateActive - the job is in "Active" state
+- BackupStateSucceeded - a final state reached when the backup job has succeeded
+- BackupStateTimeout - a final state meaning that artifact backup has been partially done
+- BackupStateError - a final state reached when the job ends with a non-zero exit code
+
+### On the spoke ###
+
+The backup workload generates an utility called `upgrade-recovery.sh` and pass the `--take-backup` option to trigger backup of `/etc`, `/usr/local`, `/var/lib/kubelet` directories and `etcd database` in the recovery partition or at the recovery folder at `/var/recovery`. In addition, the active OS deployment is pinned using ostree and the standby deployments are removed.
```

Comment:
I would avoid being too specific about the low-level details here. What is the intended audience? It's likely enough to say that it writes the upgrade-recovery.sh utility to the recovery partition and takes the pre-upgrade backup.

Reply:
Rephrased

Reply:
Rephrased